lib/Dialect/FIRRTL/FIRRTLUtils.cpp

```diff
@@ -126,6 +126,71 @@ void circt::firrtl::emitConnect(ImplicitLocOpBuilder &builder, Value dst,
   builder.create<StrictConnectOp>(dst, src);
 }
 
+// This is usable any place an implicit extension might happen.  It will force
+// a disconnect of width inference corrosponding to allowing a lhs < rhs.
+Value firrtl::expandPassiveType(ImplicitLocOpBuilder &builder,
+                                FIRRTLBaseType dstType, Value rhs) {
+  auto srcFType = rhs.getType().cast<FIRRTLType>();
+  auto srcType = srcFType.dyn_cast<FIRRTLBaseType>();
+
+  assert(dstType.isPassive() && srcType.isPassive());
```

Comment:
If the dynamic cast fails, this may crash (and not in an assertion-failure way), at a glance I think this can reach later as well.

I think having this expect a Value of FIRRTLBaseType makes the most sense, not like you'll "extend" (or want to) a Ref value.  re:above comment, there's `FIRRTLBaseValue` FWIW.